nefertari/es_query.py

```diff
@@ -8,6 +9,508 @@ def pop(self, index=None):
+
+
+class MissingProcessor(BaseProcessor):
+    name = 'missing_processor'
+
+    @staticmethod
+    def rule(term):
+        return term.value == '_missing_'
+
+    @staticmethod
+    def next():
+        return FinalProcessor
+
+    @staticmethod
+    def apply(term):
+        term.type = 'missing'
+        term.value = term.field
+        term.field = 'field'
+
+
+class MatchProcessor(BaseProcessor):
+    name = 'match_processor'
+
+    @staticmethod
+    def rule(term):
+        return (' ' in term.value and not RangeProcessor.rule(term)) or '_all' in term.field
+
+    @staticmethod
+    def next():
+        return WildCardProcessor
+
+    @staticmethod
+    def apply(term):
+        term.type = 'match'
+
+
+class TypeProcessor(BaseProcessor):
+    query_params = {'match': 'query', 'term': 'value', 'wildcard': 'value'}
+    name = 'type_processor'
+
+    @staticmethod
+    def rule(term):
+        return True
+
+    @staticmethod
+    def apply(term):
+        if term.type is None:
+            term.type = 'term'
+
+        term.query_param = TypeProcessor.query_params.get(term.type)
+
+    @staticmethod
+    def next():
+        return BoostProcessor
+
+
+class FinalProcessor(BaseProcessor):
+    name = 'final_processor'
+
+    @staticmethod
+    def rule(term):
+        return True
+
+    @staticmethod
+    def apply(term):
+        pass
+
+    @staticmethod
+    def next():
+        return None
+
+
+class Term:
+
+    def __init__(self, field, value):
+        self.field = field
+        self.value = value
+        self.type = None
+        self.processors = []
+
+    def build(self):
+        if self.field == 'should':
+            return {self.type: {self.field: self.value, 'minimum_should_match': 1}}
+        return {self.type: {self.field: self.value}}
+
+    def apply_processors(self):
+        processors_order = list(self.build_chain())
+
+        for processor_name in processors_order:
+            processor_name.apply(self)
+
+    def build_chain(self):
+        graph = dict()
+        for processor in self.processors:
+            next_processor = self.find_next_processor(processor)
+
+            if next_processor:
+                graph[processor] = next_processor
+
+        return reversed([item for item in self.find_next_vertex(graph)])
+
+    def find_next_processor(self, processor):
+        next_processor = processor.next()
+
+        if not next_processor:
+            return None
+
+        if next_processor in self.processors:
+            return next_processor
+
+        return self.find_next_processor(next_processor)
+
+    @staticmethod
+    def find_next_vertex(graph, next_=FinalProcessor):
+
+        keys = list(graph.keys())
+
+        def find_conflicts(value):
+            counter = 0
+            for item in graph.values():
+                if item == value:
+                    counter += 1
+
+                if counter > 1:
+                    return True
+            else:
+                return False
```

Comment:
As Codacy highlighted, the `return False` could be directly in the function, after the `for`